File: Min Cost Climbing Stairs_easy.cpp
#define ll long long int
#define v vector<ll>
#define fr(i,a,b) for(i=a; i<b; i++)
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        ll i, n=cost.size();
        v dp(n+2);
        fr(i,3,n+2)
            dp[i]=min(dp[i-1]+cost[i-2], dp[i-2]+cost[i-3]);
        return dp[n+1];
    }
};

File: Min Cost Climbing Stairs_easy.py
'''
    Time Complexity: O(n)
    Space Complexity: O(1)
'''
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n, one_step, two_step, curr_step = len(cost), 0, 0, 0

        for i in range(n + 1):
            curr_step = min(one_step + cost[i - 1], two_step + cost[i - 2]) if i > 1 else 0
            two_step, one_step = one_step, curr_step
        return curr_step